packages/pigment-css-react/src/Box.jsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+// eslint-disable-next-line react/prop-types
+export const Box = React.forwardRef(({ as = 'div', sx, className, style, ...rest }, ref) => {
+  const Component = as;
+  // eslint-disable-next-line react/prop-types
+  const sxClass = typeof sx === 'string' ? sx : sx?.className;
+  const classes = [className, sxClass].filter(Boolean).join(' ');
+  // eslint-disable-next-line react/prop-types
+  const sxVars = sx && typeof sx !== 'string' ? sx?.vars : {};
```

Comment:
How about?

```suggestion
  // TODO remove, logic should be handled by the transpiler, needed anyway to support <div sx={{
  const sxVars = sx && typeof sx !== 'string' ? sx?.vars : {};
```

Reply:
I don't think we'll be able to support `<div sx={{}} />` in it's current form. It'll only be possible if there are no dynamic parts in the `sx` prop value, only then will we be able to replace it with `className` prop.
On the other hand, we could do something like `<styled.div sx={{}} />` which could handle both static css as well as some dynamic values (limited by whatever is allowed in the `sx` prop).

Reply:
> It'll only be possible if there are no dynamic parts in the sx prop value

Meaning depending on props? I guess this is rare enough 😄? Most of the time I use `sx` I use static values.

> we could do something like <styled.div sx={{}} />

Like @gregberge's https://xstyled.dev/. I don't know if we need to go as far as introducing a new API.